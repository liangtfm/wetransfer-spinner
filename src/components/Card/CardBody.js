import styled from "styled-components";

const CardBody = styled.div`
  align-items: center;
  display: flex;
  flex: 1 1 80%;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0;
  width: 100%;
`;

export default CardBody;
