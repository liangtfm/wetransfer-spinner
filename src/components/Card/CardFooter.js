import styled from "styled-components";

const CardFooter = styled.div`
  align-items: center;
  border-top: ${(props) => props.border || "1px solid #000000"};
  display: flex;
  flex: 0 1 20%;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0;
  width: 100%;
`;

export default CardFooter;
