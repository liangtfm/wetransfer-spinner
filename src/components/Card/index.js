import React from "react";
import styled from "styled-components";

import CardBody from "./CardBody";
import CardFooter from "./CardFooter";

const CardContainer = styled.div`
  align-items: center;
  background-color: ${(props) => props.backgroundColor || "#ffffff"};
  border: ${(props) => props.border || "1px solid #000000"};
  border-radius: 6px;
  color: ${(props) => props.color || "#000000"};
  display: flex;
  flex: 0 1 300px;
  flex-direction: column;
  height: 450px;
  justify-content: center;
  margin: 0 25px;
`;

function Card(props) {
  return <CardContainer {...props}>{props.children}</CardContainer>;
}

export default Card;
export { CardBody, CardFooter };
