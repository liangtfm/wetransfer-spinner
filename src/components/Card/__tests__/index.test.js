import React from "react";
import { render, screen } from "@testing-library/react";

import Card, { CardBody, CardFooter } from "..";

describe("Card", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Card>Card Text</Card>);

    expect(asFragment(<Card>Card Text</Card>)).toMatchSnapshot();
    expect(screen.getByText(/Card Text/i)).toBeInTheDocument();
  });

  it("renders CardBody and CardFooter correctly", () => {
    const { asFragment } = render(
      <Card>
        <CardBody>Card Body Text</CardBody>
        <CardFooter>Card Footer Text</CardFooter>
      </Card>
    );

    expect(
      asFragment(
        <Card>
          <CardBody>Card Body Text</CardBody>
          <CardFooter>Card Footer Text</CardFooter>
        </Card>
      )
    ).toMatchSnapshot();
    expect(screen.getByText(/Card Body Text/i)).toBeInTheDocument();
    expect(screen.getByText(/Card Footer Text/i)).toBeInTheDocument();
  });
});
