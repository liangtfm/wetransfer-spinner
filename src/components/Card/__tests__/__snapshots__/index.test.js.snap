// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders CardBody and CardFooter correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm eoyqaA"
  >
    <div
      class="sc-bdfBwQ hEJWIb"
    >
      Card Body Text
    </div>
    <div
      class="sc-gsTCUz Ntqgx"
    >
      Card Footer Text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm eoyqaA"
  >
    Card Text
  </div>
</DocumentFragment>
`;
